[106X] prefiring weights for UL17 in ntuple generator

MatthiesC opened the pull request:
Followed the recipe at: https://twiki.cern.ch/twiki/bin/viewauth/CMS/L1ECALPrefiringWeightRecipe#Accessing_the_UL2017_maps
No UL16 maps available yet. Needs to be addressed before UL16 ntuple production.
The above recipe requires to pull the new `L1PrefiringMaps.root` file containing the UL17 maps. This file is not yet part of CMSSW und thus needs to be downloaded manually. I implemented this in the install scripts. If you already installed the framework before this commit, do this:

`cd <your/path/to>/CMSSW_10_6_20/src/`
`git-cms-addpkg PhysicsTools/PatUtils`
`cd PhysicsTools/PatUtils/data/`
`wget --no-check-certificate https://github.com/cms-data/PhysicsTools-PatUtils/raw/master/L1PrefiringMaps.root`
`cd ../../../`
`scram b`

MatthiesC committed:
prefiring weights for UL17 in ntuple generator

MatthiesC commented:
Here a control plot. Prefiring weights are filled.
Command:
`cmsRun ntuplewriter_mc_UL17.py maxEvents=10000 outputFile=testNtuple.root wantSummary=1`
However, I ran it with this file and not the one given in `ntuplewriter_mc_UL17.py`:
`dcap://dcache-cms-dcap.desy.de//pnfs/desy.de/cms/tier2/store/mc/RunIISummer19UL17MiniAOD/TT_Mtt-700to1000_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mc2017_realistic_v6-v1/100000/E2FD2886-F5DD-3A43-8D03-16A2B9EF5C5D.root`
The file given in `ntuplewriter_mc_UL17.py` wasn't openable due to some symbol lookup error. Full print-out with error message at the end, see below.
---> Edit Oct 7, 2021: The issue arose from not setting up the grid environment... there was no issue with the sample missing.

![prefiringWeights_test](https://user-images.githubusercontent.com/33325336/119111815-b9dfb680-ba23-11eb-955e-92d29fddbe31.png)

```
cmsRun ntuplewriter_mc_UL17.py maxEvents=10000 outputFile=testNtuple.root wantSummary=1
***  Adding fatjets_subjets for ak8CHSJetsFat
  Adding groomed genjets: ak8GenJetsSoftDrop
  Adding ungroomed genjets: ak8GenJetsFat
  Adding ungroomed jets: patJetsAk8CHSJetsFat
  Adding groomed jets: patJetsAk8CHSJetsSoftDrop
  Adding groomed jets' subjets: patJetsAk8CHSJetsSoftDropSubjets
  Adding groomed jets + subjets packer: patJetsAk8CHSJetsSoftDropPacked
***  Adding fatjets_subjets for ak8PuppiJetsFat
  Adding groomed genjets: ak8GenJetsSoftDrop
  Adding ungroomed genjets: ak8GenJetsFat
  Adding ungroomed jets: patJetsAk8PuppiJetsFat
  Adding groomed jets: patJetsAk8PuppiJetsSoftDrop
  Adding groomed jets' subjets: patJetsAk8PuppiJetsSoftDropSubjets
  Adding groomed jets + subjets packer: patJetsAk8PuppiJetsSoftDropPacked
Switching PV collection for PF2PAT: cms.InputTag("offlineSlimmedPrimaryVertices")
***********************************
Adding patJetsAK8PFPUPPI
Adding patJetsAK8PFCHS
customizePuppiTune_cff::UpdatePuppiTuneV15: Recomputing PUPPI with Tune v15, slimmedJetsPuppi and slimmedMETsPuppi
Renaming 'updatedPatJetsTransientCorrectedSlimmedJetsPuppiNewDFTraining' to 'jetsAk4Puppi'
Renaming 'updatedPatJetsTransientCorrectedSlimmedJetsNewDFTraining' to 'jetsAk4CHS'
Renaming 'updatedPatJetsTransientCorrectedRekeyPatJetsAK8PFPUPPINewDFTraining' to 'jetsAk8Puppi'
Renaming 'patJetsAK8PFCHS' to 'jetsAk8CHS'
Renaming 'updatedPatJetsTransientCorrectedRekeyPackedPatJetsAk8PuppiJetsNewDFTraining' to 'jetsAk8PuppiSubstructure'
Renaming 'updatedPatJetsTransientCorrectedPackedPatJetsAk8CHSJetsNewDFTraining' to 'jetsAk8CHSSubstructure'
Failed open file in the dCache.
%MSG-w StorageFactory::stagein():  PoolSource:source@sourceConstruction  20-May-2021 23:24:52 CEST pre-events
Failed to stage in file 'dcap://dcache-cms-dcap.desy.de//pnfs/desy.de/cms/tier2/store/mc/RunIISummer19UL17MiniAOD/TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mc2017_realistic_v6-v2/30000/BE7AE880-FE10-334A-89DC-35CCB590D9DD.root' because:
An exception of category 'FileStageInError' occurred while
   [0] Calling DCacheStorageMaker::stagein()
Exception Message:
Cannot stage in file 'dcap://dcache-cms-dcap.desy.de//pnfs/desy.de/cms/tier2/store/mc/RunIISummer19UL17MiniAOD/TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mc2017_realistic_v6-v2/30000/BE7AE880-FE10-334A-89DC-35CCB590D9DD.root', error was: "No such file or directory" (dc_errno=30)

%MSG
20-May-2021 23:24:52 CEST  Initiating request to open file dcap://dcache-cms-dcap.desy.de//pnfs/desy.de/cms/tier2/store/mc/RunIISummer19UL17MiniAOD/TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mc2017_realistic_v6-v2/30000/BE7AE880-FE10-334A-89DC-35CCB590D9DD.root
Failed open file in the dCache.
20-May-2021 23:24:52 CEST  Fallback request to file root://xrootd-cms.infn.it//store/mc/RunIISummer19UL17MiniAOD/TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mc2017_realistic_v6-v2/30000/BE7AE880-FE10-334A-89DC-35CCB590D9DD.root
cmsRun: symbol lookup error: /cvmfs/cms.cern.ch/slc7_amd64_gcc700/cms/cmssw/CMSSW_10_6_20/lib/slc7_amd64_gcc700/pluginUtilitiesXrdAdaptorPlugin.so: undefined symbol: _ZN5XrdCl3Env9PutStringERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES8_
```

mschrode commented:
Hi @MatthiesC , for some reason, all [test runs](https://gitlab.cern.ch/mschrode/UHH2-integration/-/pipelines/2631871) *with* the PR failed. The errors seem all to be different, sometimes related to some python version, I'm not sure what's going on. Also 2018 fails, even though your PR shouldn't touch/affect it, no? Could you check if you can run locally with PR eg the [2018 test run](https://github.com/UHH2/UHH2/blob/RunII_106X_v1/core/python/ntuplewriter_mc_UL18.py)?

MatthiesC commented:
Thanks for the feedback. I don't see why my commit should cause any problems. I will have a second look into it, but not today anymore.
Running UL18 locally works fine

MatthiesC commented:
@mschrode  According to the error message at [1], an error occurs in `/cvmfs/cms.cern.ch/slc7_amd64_gcc700/cms/cmssw/CMSSW_10_6_20/python/FWCore/GuiBrowsers/ConfigToolBase.py(48): __init__`:
```
class ConfigToolBase(object) :

    """ Base class for PAT tools                                                                                                                                                                                                                                                                                              
    """
    _label="ConfigToolBase"
    _defaultValue="No default value. Set parameter value."
    _path = ""
    def __init__(self):
        self._parameters=dicttypes.SortedKeysDict()
        self._description=self.__doc__
        self._comment = ''
        self.parAccepted=True
        saveOrigin(self,1)
        self._path = path.realpath(self._filename)
        self._path = self._path.split("/src/")
        self._path = self._path[1].replace("/python","") ##### ERROR OCCURS HERE: out of range, i.e. _path did not have /src/ in it; see line above #####
        #self._path = "".join(self._path)                                                                                                                                                                                                                                                                                     
        self._path = self._path.replace("/",".")
        self._path = self._path.replace(".py","")
```

It could be that this error is related to the PhysicsTools package which I added manually in the install scripts, leading to different pathing in contrast to when the PhysicsTools package would be included from `/cvmfs/cms.cern.ch/slc7_amd64_gcc700/cms/cmssw/CMSSW_10_6_20/python/......`.
But could it be that this an issue solely related to UHH2-integration? Locally, everything works

But I have no clue why that second error ([2]) occurs.

[1] https://gitlab.cern.ch/mschrode/UHH2-integration/-/jobs/13980448#L1632
[2] https://gitlab.cern.ch/mschrode/UHH2-integration/-/jobs/13980453#L188

MatthiesC commented:
PR can be closed in favor of https://github.com/UHH2/UHH2/pull/1586

MatthiesC commented:
PR closed in favor of #1586
